caldav/objects.py

```diff
@@ -1115,6 +1115,13 @@ class Todo(CalendarObjectResource):
     """
     The `Todo` object is used to represent a todo item (VTODO).
     """
+
+    def __init__(self, *args, **kwargs):
+        """Set the object's ID."""
+        super().__init__(*args, **kwargs)
+        if hasattr(self.vobject_instance.vtodo, "uid"):
```

Comment:
Presumably we'd want to handle `icalendar` data here, too?

More generally, I don't know if parsing the data in the `__init__` is the best way forward. Typically I'd implement separate factory functions for creating domain objects from raw data, rather than keeping that logic on the domain object itself.

Reply:
I'm not used to factory functions, I should probably read up on the benefits of using such

Reply:
for sure we should handle icalendar also here yes, good catch.